jukebox/src/lib.rs

```diff
@@ -123,7 +123,14 @@ impl Jukebox {
                     };
 
                     // Parse the bytes as an Hps
-                    let hps: Hps = match copy_bytes_from_file(&mut iso, real_hps_offset, hps_length)?.try_into() {
+                    let mut iso_hps = match iso.try_clone() {
+                        Ok(iso) => iso,
+                        Err(e) => {
+                            tracing::error!(target: Log::Jukebox, error = ?e, "Failed to clone iso before reading bytes. Cannot play song.");
+                            continue;
+                        },
+                    };
+                    let hps: Hps = match copy_bytes_from_file(&mut iso_hps, real_hps_offset, hps_length)?.try_into() {
```

Comment:
@ryanmcgrath @DarylPinto would like your thoughts on this change

Reply:
Discussed some more in Discord, but we've opted to address this by marking the return type of `create_offset_locator_fn` as `'static` to indicate that it does not capture any references: https://github.com/project-slippi/slippi-rust-extensions/blob/8d5bdfcaeb61327ed501cfd6b2be2055b005ed37/jukebox/src/disc/mod.rs#L57

Details about this can be found here: https://blog.rust-lang.org/2024/09/05/impl-trait-capture-rules/